src/jit/hwintrinsicxarch.cpp

```diff
@@ -153,6 +189,11 @@ InstructionSet HWIntrinsicInfo::lookupIsa(const char* className)
     {
         return InstructionSet_LZCNT;
     }
+    else if (strcmp(className, "X64") == 0)
+    {
```

Comment:
For x64 intrinsics, this means that we will check all of the candiates first, find x64 as last resort and then we look at all of them again.

Would it be better to have the check for x64 first, and then always look at all the candidates just once?

Reply:
I suggested above that we shortcut this with `if (enclosingClassName == nullptr)`.

If we have an enclosing class, it has to be `x64`, otherwise it has to be one of the other candidates.

Reply:
Will change to check for x64 first. 

@tannergooding With the latest code, `enclosingClassName` could be either `nullptr` or `""`.